model/project_ref.go

```diff
@@ -2203,6 +2242,40 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 	return res, nil
 }
 
+// ValidateProjectCreation returns an error if the total or owner/repo project limit set by the admin settings
+// has been reached. If the project is not enabled, it will grip a warning but not error.
+func ValidateProjectCreation(config *evergreen.Settings, projectRef *ProjectRef) error {
+	catcher := grip.NewBasicCatcher()
+	allEnabledProjects, err := GetNumberOfEnabledProjects()
+	if err != nil {
+		return errors.Wrap(err, "getting number of projects")
+	}
+	if allEnabledProjects >= config.ProjectCreation.TotalProjectLimit {
+		catcher.Errorf("total project limit of %d reached", config.ProjectCreation.TotalProjectLimit)
+	}
+	if !config.ProjectCreation.IsExceptionToRepoLimit(projectRef.Owner, projectRef.Repo) {
+		enabledOwnerRepoProjects, err := GetNumberOfEnabledProjectsForOwnerRepo(projectRef.Owner, projectRef.Repo)
+		if err != nil {
+			return errors.Wrapf(err, "getting number of projects for '%s/%s'", projectRef.Owner, projectRef.Repo)
+		}
+		if enabledOwnerRepoProjects >= config.ProjectCreation.RepoProjectLimit {
+			catcher.Errorf("owner repo limit of %d reached for '%s/%s'", config.ProjectCreation.RepoProjectLimit, projectRef.Owner, projectRef.Repo)
+		}
+	}
+
+	// If the project is not enabled, warn the user while still allowing project creation.
```

Comment:
re https://github.com/evergreen-ci/evergreen/pull/6181#discussion_r1081957527 this doesn't actually warn the user, it just warns us. Is this error worth logging? It seems more worth it to return an error if you try to enable a project that's over the limit. There are two things we want to do:

1) At project creation, the project is _always_ disabled, so we just want to show users a warning on the UI that they won't be able to enable it since these checks aren't met.

2) When trying to enable a project, we want to check the above constraints and return an error.
Maybe we want this function to return the error no matter what, so then we can use it for (1) to create the project, but check projectRef.Enabled and return a potential warning, and (2) to enable the project, but check projectRef.Enabled and return an error.

Reply:
return a boolean to see if we want to error or warn
will handle repo projects in another ticket
creating will also be handled in a different ticket

Reply:
returns a bool to see if we want to error or warn
now validates saving project for enabling validation.